### DISCUSSION TITLE
Variable inside a variable? Order of script steps?

### DISCUSSION THREAD

**Discussion by executive-of-dysfxn** [SilentVoid13/Templater#795] [Help] [answered]
Can I have a variable inside another variable? I know I'm trying to do things without full understanding Templater or Javascript, but I don't know what I don't know until I try things.

I thought I could have a template include a specific sentence using `suggester` and setting a yes or no option. If yes, I want to include a sentence that needs a variable inside it. If no, the sentence won't appear (I put a flag emoji as a placeholder for the moment).

Here's what I tried:

```
<%*
PairedVisit = tp.system.suggester(["Yes Paired", "Not Paired"], ["Our conversation was joined by <% isMx %><% PersonName %>.", "🚩"])
PersonName = await tp.system.prompt("Joined Convo Name")
isMx = await tp.system.suggester(["Mr", "Mrs", "Ms"], ["Mr. ", "Mrs. ", "Ms. "])
-%>
<b>CONSULT NOTE</b>
<% PairedVisit %>
```

I thought I'd be prompted to select Yes/No for `PairedVisit` first, which would add the text to the note, and then I could define `PersonName` and their title (`isMx`). Instead I get the following new file:

```
<b>CONSULT NOTE</b>
Our conversation was joined by <% isMx %><% PersonName %>.
```

or if I pick "Not Paired"

```
<b>CONSULT NOTE</b>
🚩
```

So, clearly I'm not understanding the order of operations and I'm not sure if this is too complex for my skills right now.

**Comment by welpdx** [comment, marked as answer]
Try out this bad boi:
```js
<%*
var PairedVisit = await tp.system.suggester(["Yes Paired", "Not Paired"], [true,false])
var output = `<b>CONSULT NOTE</b>\n`
if (PairedVisit) {
	var PersonName = await tp.system.prompt("Joined Convo Name")
	var isMx = await tp.system.suggester(["Mr", "Mrs", "Ms"], ["Mr. ", "Mrs. ", "Ms. "])
	output += "Our conversation was joined by " + isMx + PersonName +"."
} else if (!PairedVisit) {
	output += "🚩"
} else {
	new Notice("Exited")
	output = ""
}
-%>
<% output %>
```

<details>
           <summary style="font-size:20">Demo</summary>
           <img src="https://i.imgur.com/auivHqN.gif" alt="demo" />
</details>


### Note
Fyi, `suggester` gets the index of the selection from the first list and uses that index to get the value for the second list. This value is saved as the output of `suggester`. (e.g., if I select "Yes Paired" the output is `true`.)

Since this use case has 2 options, I just have the output as Boolean. The `if else` afterwards checks the output of suggester: 1) if true, ask for more info 2) if false, flag. 

I put the bold html tag in the output so that if you exited out of the dialogue, you don't get any output. 

I sorta changed how you intended it to work. Feel free to change it back to fit your wants. The main thing you wanted is there. 

Edit: Add note. Edited note.

**Reply by executive-of-dysfxn** [reply]
Thank you so much! Work has been so busy that I haven't been able to play with this. I will give it a try and see how things go!
